File: subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        s = 0
        count = 0
        ht = {}
        for i in nums:
            if s not in ht:
                ht[s]=1
            else:
                ht[s]+=1
            s+=i
            if s-k in ht:
                count+=ht[s-k]
        return count

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> hash = new HashMap<>();
        int sum = 0;
        int count = 0;
        for(int i = 0; i<nums.length; i++){
            sum+=nums[i];
            if(sum==k) count++;
            count += hash.getOrDefault(sum-k, 0);
            hash.put(sum, hash.getOrDefault(sum, 0)+1);
        }
        return count;
    }
}